src/Autofac/Core/Lifetime/LifetimeScope.cs

```diff
@@ -48,7 +48,7 @@ public class LifetimeScope : Disposable, ISharingLifetimeScope, IServiceProvider
         /// Protects shared instances from concurrent access. Other members and the base class are threadsafe.
         /// </summary>
         private readonly object _synchRoot = new object();
-        private readonly ConcurrentDictionary<Guid, object> _sharedInstances = new ConcurrentDictionary<Guid, object>();
+        private readonly ConcurrentDictionary<(Guid, Guid?), object> _sharedInstances = new ConcurrentDictionary<(Guid, Guid?), object>();
```

Comment:
Just as a heads up: There's no binding for `net461` and requires the package [System.ValueTuple](https://www.nuget.org/packages/System.ValueTuple/) to be installed when targeting `net461`. It must be included as conditional package-reference.

Reply:
@alsami, we've already got code using ValueTuples in the 5.0 release ([example](https://github.com/autofac/Autofac/blob/develop/src/Autofac/Util/TypeExtensions.cs#L42)).

Is there something else that keeps this working? Perhaps the fact we're also targeting `netstandard2.0`?

Reply:
@alistairjevans yeah, I was wondering why the pipeline succeeded on this PR: Could be the GAC or. Usually you get a compile error on `net461` but I happened to see that problem in other solutions as well.

Here the [issue](https://github.com/dotnet/roslyn/issues/22903) stating that this was a mistake when they've shipped it and that it has been fixed with `net471` (where the additional include is not required).

Reply:
Hmm; can you raise a separate issue for this? We should probably look into it away from this discussion.

Reply:
Okay, will do.